test/acceptance/client.js

```diff
@@ -612,27 +654,15 @@ describe('The node.js OC client', function(){
         it('should contain the error details', function(){
 
           var expectedRequestWithExtraParams = {
-            url: 'http://localhost:1234',
-            method: 'post',
+            url: 'http://localhost:1234/hello-world/~1.0.0??hi=john',
```

Comment:
Ok, it this is only for escaping that it is fine. Perhaps, just to be 100% sure about what happens, we may create a test for a component with parameters and test the response has the correct href with the structure `component/version/?params` - that could be done without regex and would be useful for tests readability I think?